src/gp/src/GPMSA.C

```diff
@@ -1406,13 +1409,16 @@ GPMSAFactory<V, M>::setUpHyperpriors()
       *(this->totalMins),
       *(this->totalMaxs)));
 
-  this->priors[0] = &(this->m_parameterPrior);
-  this->priors[1] = this->m_emulatorMean.get();
-  this->priors[2] = this->m_emulatorPrecision.get();
-  this->priors[3] = this->m_emulatorCorrelationStrength.get();
-  this->priors[4] = this->m_discrepancyPrecision.get();
-  this->priors[5] = this->m_discrepancyCorrelationStrength.get();
-  this->priors[6] = this->m_emulatorDataPrecision.get();
+  this->priors.push_back(&(this->m_parameterPrior));
+
+  if (this->num_svd_terms < numOutputs)
+    this->priors.push_back(this->m_truncationErrorPrecision.get());
```

Comment:
If this new `truncationErrorPrecision` term is being sampled, then shouldn't we still need the initial state for the Markov chain set, and the proposal covariance matrix element set?